### PR TITLE
Standalone authentication service

### PR DESCRIPTION
**Pull request by baha-a**
- #1

### CODE REVIEW

File: src/slices/authentication.slice.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  userLogin: '',
+};
+
+export const loginUser = createAsyncThunk('auth/loginUser', async (args: { login: string; pass: string }) =>
+  AuthenticationService.login(args.login, args.pass),
+);
+
+export const logoutUser = createAsyncThunk('auth/logoutUser', async () => AuthenticationService.logout());
+
+export const loginAnalytics = createAsyncThunk('auth/loginAnalytics', async () => AuthenticationService.loginAnalytics());
```
[the remark this answers is not in the document]
Comment (baha-a) [reply src/slices/authentication.slice.ts]: Removed

File: src/slices/authentication.slice.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  authenticationFailed: false,
+  userAuthorities: [],
+  userLogin: '',
+};
+
+export const loginUser = createAsyncThunk('auth/loginUser', async (args: { login: string; pass: string }) =>
+  AuthenticationService.login(args.login, args.pass),
+);
+
+export const logoutUser = createAsyncThunk('auth/logoutUser', async () => AuthenticationService.logout());
```
[the remark this answers is not in the document]
Comment (baha-a) [reply src/slices/authentication.slice.ts]: Removed

File: src/services/authentication.service.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import http from './http.service';
+
+class AuthenticationService {
+  login(login: string, password: string): Promise<void> {
+    return http.post('/cs-login', { login, password });
+  }
+
+  loginAnalytics(): Promise<void> {
```
[the remark this answers is not in the document]
Comment (baha-a) [reply src/services/authentication.service.ts]: Removed

File: src/services/authentication.service.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    return http.post('/cs-login-analytics', {});
+  }
+
+  loginWithTaraJwt(): Promise<void> {
+    return http.post('/cs-login-with-tara-jwt');
+  }
+
+  logout(): Promise<void> {
+    return http.post('/cs-logout');
```
[the remark this answers is not in the document]
Comment (baha-a) [reply src/services/authentication.service.ts]: Removed

File: package.json [hunk 5]
```diff
@@ -0,0 +1,85 @@
+{
+  "name": "byk-customer-service",
```
[the remark this answers is not in the document]
Comment (baha-a) [reply package.json]: Renamed

**Review by 1AhmedYasser — APPROVED** [review]
LGTM

**Pull request merged**